raster-foundry/raster-foundry
# Add lambda function for Landsat 8 SNS processing

## Overview

This PR adds a `handle` function for creating landsat 8 scene from the Landsat 8 SNS topic.

### Checklist

- [x] PR opened to fix post body for authentication from python client (https://github.com/raster-foundry/raster-foundry-python-client/pull/70)

### Notes

It also exposed an issue with the python client where attempts to get an API token with a refresh token fail because the body is wrong. I added that in the checklist and will ilnk the PR when I open it.

Also, there are automated tests here, but until we know where this is going to live I didn't think it was worthwhile to do the work to ensure that `tox` runs in `app-lambda` in ci.

## Testing Instructions

 * cd to `app-lambda`
 * `python3.6 -m venv venv`
 * `source venv/bin/activate`
 * `pip install -r requirements.txt`
 * `tox` -- should pass
 * then -- `export RF_API_TOKEN='a valid api token'`
 * `export RF_API_HOST=localhost:9091`
 * re-assemble the api server
 * bring up your frontend and servers
 * start an ipython session in the virtual environment
 *
```python
import json
test_event = json.loads("""<copy and paste from linked issue>""")
from rflambda.landsat8.handle import handle
handle(test_event, {})
```
 * it should create a new scene in northeastern greeland for August 5, 2018
 * you should be able to preview that scene's thumbnails on the map
 * and in the scene detail modal (check out the insane quantity of metadata available from the MTL json)
 * and you should be able to download individual images from it
 
Closes #4114
